tests/pass-dep/shims/libc-misc.rs

```diff
@@ -429,6 +429,19 @@ fn test_strncpy() {
         assert_eq!(*slc, [1i8, 2, 3, 0]);
         libc::free(dest);
     }
+
+    // case: bytes without `\0` ending, and followed by uninitialized bytes
+    unsafe {
+        let slc = [1i8, 2, 3];
+        let src = libc::malloc(5);
+        libc::memcpy(src, slc.as_ptr() as *const libc::c_void, 3);
+        let dest = libc::calloc(4, 1);
+        libc::strncpy(dest as *mut libc::c_char, src as *const libc::c_char, 3);
```

Comment:
The issue I'm worried about is if the `n` here is larger . So like if you use 10 in strncpy, but the 4th byte is uninit